src/backend/cpu/anisotropic_diffusion.cpp

```diff
@@ -0,0 +1,34 @@
+/*******************************************************
+ * Copyright (c) 2017, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <af/dim4.hpp>
+#include <Array.hpp>
+#include <anisotropic_diffusion.hpp>
+#include <kernel/anisotropic_diffusion.hpp>
+
+namespace cpu
+{
+template<typename T>
+void anisotropicDiffusion(Array<T>& inout, const float dt,
+                          const float mct, const af_flux_function fftype,
+                          const af::diffusionEq eq)
+{
+    if (eq==AF_DIFFUSION_MCDE)
+        getQueue().enqueue(kernel::anisotropicDiffusion<T, true>, inout, dt, mct, fftype);
+    else
+        getQueue().enqueue(kernel::anisotropicDiffusion<T, false>, inout, dt, mct, fftype);
+}
+
+#define INSTANTIATE(T)\
+template void anisotropicDiffusion<T>(Array<T> &inout, const float dt, const float mct,\
+                                      const af_flux_function fftype, const af::diffusionEq eq);
```

Comment:
its odd you are using both c and c++ enums here. you should try to be consistent.